src/topologies/halfedge.jl

```diff
@@ -144,126 +144,116 @@ function HalfEdgeTopology(elems::AbstractVector{<:Connectivity}; sort=true)
 
   # sort elements to make sure that they
   # are traversed in adjacent-first order
-  adjelems = sort ? adjsort(elems) : elems
-  eleminds = sort ? indexin(adjelems, elems) : 1:length(elems)
+  elemsort = sort ? adjsort(elems) : elems
+  eleminds = sort ? indexin(elemsort, elems) : 1:length(elems)
+  adjelems::Vector{Vector{Int}} = map(collect ∘ indices, elemsort)
 
   # start assuming that all elements are
-  # oriented consistently as CCW
-  CCW = trues(length(adjelems))
+  # oriented consistently (e.g. CCW)
+  isreversed = falses(length(adjelems))
 
-  # initialize with first element
+  # initialize half-edges using first element
   half4pair = Dict{Tuple{Int,Int},HalfEdge}()
-  elem = first(adjelems)
-  inds = collect(indices(elem))
-  v = CircularVector(inds)
-  n = length(v)
-  for i in 1:n
-    half4pair[(v[i], v[i + 1])] = HalfEdge(v[i], eleminds[1])
+  elem = first(eleminds)
+  inds = first(adjelems)
+  n = length(inds)
+  for i in eachindex(inds)
+    u = inds[i]
+    v = inds[mod1(i + 1, n)]
+    # insert halfedges u -> v and v -> u
+    he = get!(() -> HalfEdge(u, elem), half4pair, (u, v))
+    half = get!(() -> HalfEdge(v, nothing), half4pair, (v, u))
+    he.half = half
+    half.half = he
   end
 
-  # insert all other elements
-  for e in 2:length(adjelems)
-    elem = adjelems[e]
-    inds = collect(indices(elem))
-    v = CircularVector(inds)
-    n = length(v)
-    for i in 1:n
-      # if pair of vertices is already in the
-      # dictionary this means that the current
-      # polygon has inconsistent orientation
-      if haskey(half4pair, (v[i], v[i + 1]))
-        # delete inserted pairs so far
-        CCW[e] = false
-        for j in 1:(i - 1)
-          delete!(half4pair, (v[j], v[j + 1]))
-        end
-        break
+  # update half-edges using all other elements
+  added = false
+  disconnected = false
+  remaining = collect(2:length(adjelems))
+  while !isempty(remaining)
+    iter = 1
+    while iter ≤ length(remaining)
+      other = remaining[iter]
+      elem = eleminds[other]
+      inds = adjelems[other]
+      n = length(inds)
+      if anyhalf(inds, half4pair) || disconnected
+        # at least one half-edge has been found, so we can assess
+        # the orientation w.r.t. previously added elements/edges
+        added = true
+        disconnected = false
+        deleteat!(remaining, iter)
       else
-        # insert pair in consistent orientation
-        half4pair[(v[i], v[i + 1])] = HalfEdge(v[i], eleminds[e])
+        iter += 1
+        continue
```

Comment:
Could you please confirm this `continue` is only skipping the rest of the inner `while` loop?

Reply:
That's correct. `break` and `continue` only affect the whichever loop is most local to the `break`/`continue` call.